R/assertions.R

```diff
@@ -595,13 +602,14 @@ assert_integer_scalar <- function(arg, subset = "none", optional = FALSE) {
   }
 
   if (!is_integerish(arg) || length(arg) != 1L || !is.finite(arg) || !eval(subsets[[subset]])) {
```

Comment:
Could we have a different message for the case that the argument is an integer but not in the subset?
```
> assert_integer_scalar(2L, subset = "negative")
Error:
! Argument `2L` must be a negative integer scalar, but is an integer.
Run `rlang::last_trace()` to see where the error occurred.
```
doesn't read well.
I would prefer something like
```
> assert_integer_scalar(2L, subset = "negative")
Error:
! Argument `2L` must be negative, but is non-negative.
Run `rlang::last_trace()` to see where the error occurred.
```